Package.swift

```diff
@@ -27,29 +27,41 @@ let package = Package(
     ],
     dependencies: [
         .package(url: "https://github.com/StanfordSpezi/SpeziFoundation", from: "1.0.2"),
-        .package(url: "https://github.com/StanfordBDHG/XCTRuntimeAssertions", from: "1.0.1")
+        .package(url: "https://github.com/StanfordBDHG/XCTRuntimeAssertions", from: "1.0.1"),
+        .package(url: "https://github.com/apple/swift-collections.git", from: "1.1.1"),
+        .package(url: "https://github.com/realm/SwiftLint.git", .upToNextMinor(from: "0.55.1"))
```

Comment:
Wondering if it rather makes sense to add this to the UI Testing app and not the main Swift Package. Building Swift Syntax takes a lot of time and there are also a decent amount of dependencies in SwiftLint that we might not want to pull in any projects relying on Spezi.

Swift collections sounds good to me, scoped and clearly added for the core functionality.

Wondering if we should rather enable the strict concurrency checks as documented in #103.

Reply:
The idea was to integrate SwiftLint directly into Xcode and the development workflow when developing for Spezi frameworks. Like modern IDEs have it since decades 🙂
Smartest move would be to have the build tool rely on a pre-compiled binary (like every other set up does it) and seems like they also acknowledge the issue as for macOS targeted builds they just rely on a pre-built binary (see [here](https://github.com/realm/SwiftLint/blob/3b19e7d99b3a91ed95862659d0c306ca970126f0/Package.swift#L172-L178)). There have also been issues with swiftlint as they tend to specify exact version ranges (e.g., https://github.com/realm/SwiftLint/issues/5533, https://github.com/jpsim/SourceKitten/pull/799) which with the current version was fixed but they already introduced an exact version dependency with the upcoming release.

Your approach of just adding it to the UI Testing app would only lint source files within the testing application, not its dependencies.

And you are right with the build times, just verified this by data current build from this PR took [1m 24s](https://github.com/StanfordSpezi/Spezi/actions/runs/9636148663/job/26573936418?pr=106) and a previous build from the main branch took just [39s](https://github.com/StanfordSpezi/Spezi/actions/runs/9384845089/job/25841560053). 

To summarize, I agree that there are very good points to avoid including swiftlint as a dependency. I think I also tend to not include it for every project.

One approach which could be worth exploring, is including it only if a certain environment variable is set, similar to how firebase controls building their dependencies from source https://github.com/firebase/firebase-ios-sdk/blob/09920d156113a6c32916fd8056ee14b6c05bc434/Package.swift#L1419-L1424. You would need to manually start Xcode from command line once but it would persist as long as this instance is running. The build time would stay low by default with a somewhat easy one shot command to enter "developer mode" for a day (or as long Xcode is running).

Reply:
Added Strict Concurrency checking with the latest commit 👍 
Further I introduced the `SPEZI_DEVELOPMENT_SWIFTLINT` environment variable that conditionally allows to define swiftlint as a dependency.

Reply:
This PR technically supersedes #104 now. This PR additionally also fixes the warnings. However, we currently do not mark warnings as errors.

Reply:
100% agree; a quite unfortunate state at this point 🤷‍♂️

I like the idea of having a environment variable to address this issue and allowing us to easily enable SwiftLint conditionally within an dev environment 👍

Thanks for fixing this here @Supereg! We should then use #104 for marking warnings as errors @pauljohanneskraft, that is something that would be great independent of the exact warnings fixed here and would be great as an addition to the overall build pipeline.